Remove tomli

cxzhong opened the pull request:
tomli is a TOML parser that was needed for Python < 3.11. Since Python 3.11+, tomllib is part of the standard library. tomli-w (TOML writer) is kept as it's still needed for writing TOML files.

Changes:
- Removed build/pkgs/tomli directory
- Updated package dependencies to remove tomli references
- Remove the unused 3.10 env file

<!-- ^ Please provide a concise and informative title. -->
<!-- ^ Don't put issue numbers in the title, do this in the PR description below. -->
<!-- ^ For example, instead of "Fixes #12345" use "Introduce new method to calculate 1 + 2". -->
<!-- v Describe your changes below in detail. -->
<!-- v Why is this change required? What problem does it solve? -->
<!-- v If this PR resolves an open issue, please link to it here. For example, "Fixes #12345". -->



### :memo: Checklist

<!-- Put an `x` in all the boxes that apply. -->

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

<!-- List all open PRs that this PR logically depends on. For example, -->
<!-- - #12345: short description why this is a dependency -->
<!-- - #34567: ... -->

cxzhong committed:
Remove tomli dependency (keep tomli-w)

tomli is a TOML parser that was needed for Python < 3.11.
Since Python 3.11+, tomllib is part of the standard library.
tomli-w (TOML writer) is kept as it's still needed for writing TOML files.

Changes:
- Removed tomli from all environment-*.yml files
- Removed build/pkgs/tomli directory
- Updated package dependencies to remove tomli references

cxzhong committed:
Delete environment-3.10-win.yml

tobiasdiez reviewed (changes requested):
The conda env files are automatically generated. So if tomli is in there, it means that another python library still needs it (or the metadata in some conda package needs to be updated to reflect that tomli is no longer needed).

Cannot say much about the changes in `build`. If all of those pkgs are no longer actually using tomli, then it would be fine to remove it.

cxzhong committed:
reupload env files, and try to exclude the tools dependencies conda-lock from env

cxzhong committed:
Update the env files

cxzhong committed:
update the env files

cxzhong committed:
Update cache key format in CI workflow to include Python version and OS type

cxzhong committed:
Update environment configuration files for Sage development

- Updated input hashes in environment files for consistency.
- Changed version of `brial` from 1.2.3 to 1.2.12 in Linux and macOS environments.
- Updated `libbrial` version from 1.2.3 to 1.2.12 in Linux and macOS environments.
- Adjusted `linbox` version from 1.7.1 to 1.7.0 in Linux and macOS environments.
- Changed `m4ri` version from 20250128 to 20140914 in Linux and updated `m4rie` version accordingly.
- Updated various dependencies in the Windows environment, removing obsolete packages and ensuring compatibility.

cxzhong committed:
Update environment configuration and improve conda update script

- Updated `environment-3.13-win.yml` with new dependencies and version updates.
- Added several new packages including `annotated-types`, `backports`, `click`, and others.
- Updated Python version from 3.13.7 to 3.13.8 and adjusted related dependencies.
- Enhanced `update-conda.py` to use threading for processing platforms and Python versions concurrently.
- Refactored dependency retrieval logic to improve clarity and maintainability.
- Removed unnecessary comments and cleaned up the code for better readability.

cxzhong committed:
Merge origin/develop into remove-tomli (conflicts resolved preferring origin/develop)

cxzhong committed:
Sync .github/workflows/ci-meson.yml from origin/develop into remove-tomli

cxzhong committed:
Remove unnecessary newline before update_conda call

cxzhong committed:
Remove 'toml' dependency from pyproject.toml

Removed 'toml' from the list of dependencies.

tobiasdiez reviewed (approved):
LGTM, but I cannot say much about the changes in the `build` folder.
